File: examples/weather/forecast/node/forecaseWeather.js
import * as weather from "m3o/weather";

// Get the weather forecast for the next 1-10 days
async function ForecaseWeather() {
  let weatherService = new weather.WeatherService(process.env.MICRO_API_TOKEN);
  let rsp = await weatherService.forecast({
    days: 2,
    location: "London",
  });
  console.log(rsp);
}

await ForecaseWeather();

File: examples/weather/forecast/go/forecaseWeather.go
package example

import (
	"fmt"
	"github.com/micro/services/clients/go/weather"
	"os"
)

// Get the weather forecast for the next 1-10 days
func ForecaseWeather() {
	weatherService := weather.NewWeatherService(os.Getenv("MICRO_API_TOKEN"))
	rsp, err := weatherService.Forecast(&weather.ForecastRequest{
		Days:     2,
		Location: "London",
	})
	fmt.Println(rsp, err)
}
